main.go

```diff
@@ -31,8 +27,8 @@ import (
 	gormtrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/gorm.io/gorm.v1"
 )
 
-func main() {
-
+// TODO: move to service.go
+func NewService(wg *sync.WaitGroup) *Service {
```

Comment:
+ why do we need this WaitGroup passed in here. This feels a bit strange

Reply:
the service is run on a separate process. The go process exits before that process is gracefully shut down (e.g. disconnect from the relay). The app wasn't actually exiting gracefully though. I have fixed these now, maybe we can review it together?